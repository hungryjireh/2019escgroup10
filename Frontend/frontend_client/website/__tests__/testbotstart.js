import React from "react";
import { render, cleanup, fireEvent, waitForElement, getByLabelText, getByText, getByTestId } from 'react-testing-library';
import BotStart from "../components/Hero/Bot/BotStart";
import {BlueHead, BotImage, BotFormOne, BotFormTwo, BotButton} from "../components/Hero/Bot/BotStart";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

it("BlueHead renders", () => {
    const { asFragment } = render(<BlueHead />);
    expect(asFragment()).toMatchSnapshot();
});

it("BotImage renders", () => {
    const { asFragment } = render(<BotImage />);
    expect(asFragment()).toMatchSnapshot();
});

it("BotFormOne renders", () => {
    const { asFragment } = render(<BotFormOne />);
    expect(asFragment()).toMatchSnapshot();
});

it("BotFormTwo renders", () => {
    const { asFragment } = render(<BotFormTwo />);
    expect(asFragment()).toMatchSnapshot();
});

test("BotButton Components", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <BrowserRouter>
            <BotButton to="/botchatting"/>
        </BrowserRouter>,
    div
    );
    ReactDOM.unmountComponentAtNode(div);
});

test("BotStart renders", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <BrowserRouter>
            <BotStart />
        </BrowserRouter>,
    div
    );
    ReactDOM.unmountComponentAtNode(div);
});
