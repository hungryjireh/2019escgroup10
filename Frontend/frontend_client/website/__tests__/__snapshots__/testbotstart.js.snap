// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueHead renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH cbqDFY"
  />
</DocumentFragment>
`;

exports[`BotFormOne renders 1`] = `
<DocumentFragment>
  <input
    class="sc-bxivhb fuxUrh"
  />
</DocumentFragment>
`;

exports[`BotFormTwo renders 1`] = `
<DocumentFragment>
  <input
    class="sc-ifAKCX bVKKcl"
  />
</DocumentFragment>
`;

exports[`BotImage renders 1`] = `
<DocumentFragment>
  <img
    class="sc-htpNat hOunee"
  />
</DocumentFragment>
`;
